pkgs/development/compilers/cudatoolkit/redist/build-cuda-redist-package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, autoAddOpenGLRunpathHook
+}:
+
+pname:
+attrs:
+
+let
+  arch = "linux-x86_64";
+in stdenv.mkDerivation {
+  inherit pname;
+  inherit (attrs) version;
+
+  src = assert (lib.hasAttr arch attrs); fetchurl {
+    url = "https://developer.download.nvidia.com/compute/cuda/redist/${attrs.${arch}.relative_path}";
+    inherit (attrs.${arch}) sha256;
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    # This hook will make sure libcuda can be found
+    # in typically /lib/opengl-driver by adding that
+    # directory to the rpath of all ELF binaries.
+    # Check e.g. with `patchelf --print-rpath path/to/my/binary
+    autoAddOpenGLRunpathHook
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc.lib
+  ];
+
+  dontBuild = true;
+
+  # TODO: choose whether to install static/dynamic libs
+  installPhase = ''
+    runHook preInstall
+    rm LICENSE
+    mkdir -p $out
+    mv * $out
+    runHook postInstall
+  '';
+
+  meta = {
+    description = attrs.name;
+    license = lib.licenses.unfree;
```

Comment:
Oh, by the way, we need `licenses.unfreeRedistributable`! I don't remember if I have filters for `meta.license.redistributable`, but numtide/nixpkgs-unfree iirc does

Reply:
From h the previous discussions I understood that redistributable is something we desire, but seems isn't reality.

Reply:
We're literally downloading stuff from `.../redist/...` URL.
Again, the only part that is not redistributable is `libcuda.so`, which lives in `nvidia_x11`

Reply:
But! I thought I pointed at one of the expressions that were new. The `cudatoolkit` has had `unfree` license in previous revisions, so let's have that discussion in a separate PR

Reply:
I think the question is whether or not patchelf constitutes modification or not... but IANAL. We can always change the licenses in the future as necessary. I'm not sure how to get a definitive answer on these kinds of things.